cluster-agent/controllers/managed-gitops/eventloop/operation_event_loop_test.go

```diff
@@ -1196,6 +1198,32 @@ var _ = Describe("Operation Controller", func() {
 				err = k8sClient.Update(ctx, applicationCR)
 				Expect(err).To(BeNil())
 			}
+			refreshHandler := func(appCR appv1.Application, closeRefreshHandler chan struct{}) {
+				for {
```

Comment:
In the success, non-conflict case, this loop will run over and over again, as fast as the CPU will allow (since this is a mock there is no I/O bound): instead, you may want to consider add short delay after each iteration.

Reply:
I've added a delay. Also, the goroutine will be running only during the lifecycle of a test since we close it in `AfterEach`